lib/truffle/java/java_proxy.rb

```diff
@@ -25,5 +28,16 @@ def to_java_object
 
   def eql?(another)
   end
-  
+
+  def self.[](*dims)
+    ArrayJavaProxyCreator.new(java_class, *dims)
+  end
+
+  def self.inherited(thing)
```

Comment:
For inherited? It's an override of `Class#inherited` and has to be called that to be called when subclasses are created.